app/org/sagebionetworks/bridge/models/schedules/TaskScheduler.java

```diff
@@ -75,9 +75,9 @@ private void addTaskForEachActivityAtTime(List<Task> tasks, ScheduleContext cont
                         task.setExpiresOn(expiresOn);
                         task.setHidesOn(expiresOn.getMillis());
                     }
-                    if (context.getSchedulePlanGuid() != null) {
-                        task.setRunKey(BridgeUtils.generateTaskRunKey(task, context));
-                    }
+                    task.setRunKey(BridgeUtils.generateTaskRunKey(task, plan.getGuid()));
+                    task.setMinAppVersion(plan.getMinAppVersion());
+                    task.setMaxAppVersion(plan.getMaxAppVersion());
```

Comment:
Do we have a unit test to make sure the scheduler is propagating the min/max app versions to the tasks?